cvat-ui/src/components/invitation-watcher/invitation-watcher.tsx

```diff
@@ -0,0 +1,102 @@
+                invitations.push(invitationKey);
+                localStorage.setItem('invitations', JSON.stringify(invitations));
+                localStorage.removeItem('currentOrganization');
+            }
+        }
+
+        if (user && user.isVerified) {
+            const storedInvitations = localStorage.getItem('invitations') || '[]';
+            const invitations = JSON.parse(storedInvitations);
+
+            const checkRequests = invitations.map(
+                (key: string) => core.organizations.invitation(key),
+            );
+            Promise.allSettled(checkRequests).then((promises) => {
+                const validInvitations = promises
+                    .map((promise) => (promise.status === 'fulfilled' ? promise.value : null))
+                    .filter((invitation) => !!invitation);
+
+                promises.forEach((promise) => {
+                    if (promise.status === 'rejected') {
+                        notification.error({
+                            message: 'Invitation error',
+                            className: 'cvat-invitation-error',
+                            description: promise.reason.toString(),
+                        });
+                    }
+                });
+
+                const expiredInvitations = validInvitations.filter((invitation) => invitation.expired);
+                expiredInvitations.forEach((invitation) => {
+                    notification.error({
+                        message: 'Invitation error',
+                        className: 'cvat-invitation-error',
+                        description: `The invitation to ${invitation.organization.slug} is expired. ` +
+                            'Please, contact organization owner.',
+                    });
+                });
+
+                const invitationsToBeShown = validInvitations.filter((invitation) => !invitation.expired);
+                if (invitationsToBeShown.length === 0) {
+                    localStorage.removeItem('invitations');
+                }
+                invitationsToBeShown.forEach((invitation: Invitation, index) => {
+                    const slug = invitation.organization instanceof Organization ? invitation.organization.slug : '';
+                    Modal.confirm({
+                        title: `You've been invited to an organization ${slug} by ${invitation.owner?.username}`,
+                        content: 'Do you want to join?',
+                        className: 'cvat-invitaion-confirm-modal',
+                        onOk: () => {
+                            dispatch(acceptInvitationAsync(invitation.key, (orgSlug) => {
+                                localStorage.setItem('currentOrganization', orgSlug);
+                                if (index === invitationsToBeShown.length - 1) {
+                                    window.location.reload();
+                                    localStorage.removeItem('invitations');
+                                }
```

Comment:
As I remember dispatch returns promise in case with async function, so, we do not need callbacks here, just dispatch acceptInvitationAsync(invitation.key).then((orgSlug) => ...)